Add Repository readme (#13)
jenkins-infra/documentation: README.md

# Documentation

This repository's purpose is to keep track of the knowledge around the Jenkins infrastructure project.
It gathers information for the following areas:

> * "Maintenance" section contains document used to prepare service maintenance like systems upgrade.

Reviewer: ```suggestion
* "Maintenance" section contains document used to prepare service maintenances like systems upgrade.
```

* "Meeting" section contains meeting notes

> * "Service" section contains service description

Reviewer: ```suggestion
* "Service" section contains service descriptions
```

> The way contributors can collaborate in this repository depends on either we want "synchronized" or "asynchronized" communication.

Reviewer: ```suggestion
The contribution workflow depends on either you want a "synchronous" or "asynchronous" communication.
```

> In the cased of "synchronized" communication, it's better to use [hackmd.io](https://hackmd.io/team/jenkins-infra) with documents linked in the jenkins-infra channel. Once the collaboration is done, someone needs to push the document into this repository.

Reviewer: ```suggestion
In the cased of "synchronous" communication, use [hackmd.io](https://hackmd.io/team/jenkins-infra) with documents linked in the `jenkins-infra` channel. Once the collaboration is done, a maintainer needs to push the document into this repository.
```

> Feel free to request access, our goals is only to avoid spammers.

Reviewer: ```suggestion
Feel free to request access: restriction of the "push to GitHub" action is only to avoid spammers.
```

Author: On hackmd.io external contributors only have comment permission

Reviewer: OK then, I'm not sure about what do we mean by "Feel free to request access" ? Which access to what?

Author: hackmd.io => https://hackmd.io/team/jenkins-infra

> In the case of "asynchronized" communication, it's better to use the git trunk-based development as it gives more time

Reviewer: ```suggestion
In the case of "asynchronous" communication, open a [GitHub Pull Request](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests) to ask for a review
```

> to contributors to provide feedback, and it integrates better with Github notifications.

Reviewer: ```suggestion
from maintainers to provide feedback, share knowledge and benefit from the Github notifications system.
```
